### PR TITLE
Resolves #3753 - Skype ID/Google Hangouts ID can exceed database field

### LINKED ISSUE #3753: ICQ Number/Yahoo ID/Skype ID/Google Hangouts ID can exceed database field
The values of ICQ Number, Yahoo ID, Skype ID and Google Hangouts ID are not validated (beside casting ICQ to integer) so you can easily exceed the length of the associated database fields:
Quote:SQL Error: 1406 - Data too long for column 'yahoo' at row 1

Original thread: [ICQ Number/Yahoo ID/Skype ID/Google Hangouts ID can exceed database field](https://community.mybb.com/thread-223686.html)

### PR DESCRIPTION
**Pull request by Ben-MyBB**
Finally got there with the help of @yuliu 

Resolves #3753

### CODE REVIEW

File: usercp.php [hunk 1]
```diff
@@ -243,13 +243,21 @@
...
+            }
+        }
+        else
+        {
+            $user[$cfield] = $mybb->get_input($cfield);
+            if(my_strlen($user[$cfield]) > 75)
+            {
+                error($lang->contact_field_error);
+            }
+        }
```
Comment (euantorano) [inline usercp.php:260]: Indentation should be tabs:

```suggestion
		{
			$user[$cfield] = $mybb->get_input($cfield, 1);

			if(my_strlen($user[$cfield]) > 10)
			{
				error($lang->contact_field_icqerror);
			}
		}
		else
		{
			$user[$cfield] = $mybb->get_input($cfield);

			if(my_strlen($user[$cfield]) > 75)
			{
				error($lang->contact_field_error);
			}
		}
```

File: inc/languages/english/usercp.lang.php [hunk 2]
```diff
@@ -211,6 +211,8 @@
 $l['contact_field_icq'] = "ICQ Number:";
 $l['contact_field_skype'] = "Skype ID:";
 $l['contact_field_google'] = "Google Hangouts ID:";
+$l['contact_field_error'] = "Sorry, but we cannot change your contact information as the ID specified is too long. You can try to use a shorter ID number which is less than 75 characters long. This error will be caused by either the Google Hangouts ID or Skype ID.";
```
Comment (euantorano) [inline inc/languages/english/usercp.lang.php:214]: This seems a bit long, ideally error messages should be short and sweet while still being descriptive IMO.

```suggestion
$l['contact_field_error'] = "Sorry, but we cannot change your contact information as the ID specified is too long. Google Hangouts IDs and Skype IDs can be up to 75 characters long.";
```

File: inc/languages/english/usercp.lang.php [hunk 3]
```diff
@@ -211,6 +211,8 @@
...
+$l['contact_field_icqerror'] = "Sorry, but we cannot change your ICQ number as the number specified is too long. You can try to use a shorter ICQ number which is less than 10 characters long.";
```
Comment (euantorano) [inline inc/languages/english/usercp.lang.php:215]:
```suggestion
$l['contact_field_icqerror'] = "Sorry, but we cannot change your ICQ number as the number specified is too long. ICQ numbers can be up to 10 characters long.";
```

**Comment by Ben-MyBB** [comment]
Thanks @euantorano - updated.

**Comment by euantorano** [comment]
Thinking about it, I think we should move the validation into the user data handler and then let [the inline errors](https://github.com/mybb/mybb/blob/f317e37b062e08ccb26a93180d57d6d84a6c0150/usercp.php#L294) be set?

**Comment by yuliu** [comment]
> Thinking about it, I think we should move the validation into the user data handler and then let [the inline errors](https://github.com/mybb/mybb/blob/f317e37b062e08ccb26a93180d57d6d84a6c0150/usercp.php#L294) be set.

I was thinking in the same way but just saw the away date gets validated there in usercp.php. I think we can make a new PR to move them to the userHandler. Probably in next point release

**Comment by euantorano** [comment]
Ok, I'll create an issue then.

**Pull request merged**